src/isomorphic/classic/element/__tests__/ReactElement-test.js

```diff
@@ -57,35 +57,105 @@ describe('ReactElement', function() {
     expect(element.props).toEqual(expectation);
   });
 
-  it('should warn when `key` is being accessed', function() {
+  /**
+   * Check the warning about accessing props.key for correct component name.
+   *
+   * @param {object} testOpts Options for the test.
+   *   testOpts.Component {ReactClass|function} ReactComponent class or
+   *   stateless function.
+   *
+   *   testOpts.expectedName {string} The name that the component is expected to
+   *   be identified by in the warning message.
+   *
+   * @return {void}
+   */
+  function testKeyAccess(testOpts) {
+    /**
+     * Generate the expected warning message for accessing props.key.
+     *
+     * @param {object} opts Options.
+     *   opts.name {string} Component name.
+     *
+     * @return {string}
+     */
+    function getWarning(opts) {
+      return `
+        ${opts.name}: ${'`key`'} is not a prop. Trying to access it will result
+        in ${'`undefined`'} being returned. If you need to access the same
+        value within the child component, you should pass it as a different
+        prop. (https://fb.me/react-special-props)
+      `.replace(/\n */g, ' ').trim();
+    }
+
     spyOn(console, 'error');
+    var {Component, expectedName} = testOpts;
     var container = document.createElement('div');
-    var Child = React.createClass({
-      render: function() {
-        return <div> {this.props.key} </div>;
-      },
-    });
-    var Parent = React.createClass({
-      render: function() {
-        return (
-          <div>
-            <Child key="0" />
-            <Child key="1" />
-            <Child key="2" />
-          </div>
-        );
-      },
-    });
+
+    function render(props) {
+      return <div> {props.key} </div>;
+    }
+
     expect(console.error.calls.length).toBe(0);
-    ReactDOM.render(<Parent />, container);
+    ReactDOM.render(<Component key="0" render={render} />, container);
     expect(console.error.calls.length).toBe(1);
-    expect(console.error.argsForCall[0][0]).toContain(
-      'Child: `key` is not a prop. Trying to access it will result ' +
-      'in `undefined` being returned. If you need to access the same ' +
-      'value within the child component, you should pass it as a different ' +
-      'prop. (https://fb.me/react-special-props)'
-    );
-  });
+    expect(console.error.argsForCall[0][0]).toContain(getWarning({
+      name: expectedName,
+    }));
+  }
+
+  it(
```

Comment:
Having many tests like this seems unnecessary to me. If we extract `getComponentName` into a function, we can get away with one test here and a bunch of tests for specific cases in `getComponentName-test`. The indirection with `testKeyAccess` seems over-abstracted and unnecessary to me. Can we make this file simple again?